firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -145,28 +151,117 @@ public UpdateTask updateIfNewReleaseAvailable() {
-            })
-        .addOnFailureListener(
-            e -> {
-              postProgressToCachedUpdateIfNewReleaseTask(
-                  UpdateProgress.builder()
-                      .setApkFileTotalBytes(UNKNOWN_RELEASE_FILE_SIZE)
-                      .setApkBytesDownloaded(UNKNOWN_RELEASE_FILE_SIZE)
-                      .setUpdateStatus(UpdateStatus.NEW_RELEASE_CHECK_FAILED)
-                      .build());
-              setCachedUpdateIfNewReleaseCompletionError(
-                  e,
-                  new FirebaseAppDistributionException(
-                      Constants.ErrorMessages.NETWORK_ERROR,
-                      FirebaseAppDistributionException.Status.NETWORK_FAILURE));
-            });
+              return showUpdateAlertDialog(release);}
+              )
+        .onSuccessTask( unused ->
+            updateApp(true)
+                .addOnProgressListener(
+                    this::postProgressToCachedUpdateIfNewReleaseTask)
+                .addOnFailureListener(
+                    this::setCachedUpdateIfNewReleaseCompletionError))
+        .addOnFailureListener(this::setCachedUpdateIfNewReleaseCompletionError);
 
     synchronized (updateIfNewReleaseTaskLock) {
       return cachedUpdateIfNewReleaseTask;
     }
   }
 
+  private Task<Void> showSignInDialog() {
+    if (isTesterSignedIn()) {
+      return Tasks.forResult(null);
+    }
+
+    TaskCompletionSource<Void> showDialogTask = new TaskCompletionSource<>();
+
+    Activity currentActivity = lifecycleNotifier.getCurrentActivity();
+    if (currentActivity == null) {
+      LogWrapper.getInstance().e("No foreground activity found.");
+      showDialogTask.setException(
+          new FirebaseAppDistributionException(
+              ErrorMessages.APP_BACKGROUNDED,
+              FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
+    }
+
+    signInDialog = new AlertDialog.Builder(currentActivity).create();
+    Context context = firebaseApp.getApplicationContext();
+    signInDialog.setTitle(context.getString(R.string.signin_dialog_title));
+    signInDialog.setMessage(context.getString(R.string.singin_dialog_message));
+
+    signInDialog.setButton(
+        AlertDialog.BUTTON_POSITIVE,
+        context.getString(R.string.singin_yes_button),
+        (dialogInterface, i) -> showDialogTask.setResult(null));
+
+    signInDialog.setButton(
+        AlertDialog.BUTTON_NEGATIVE,
+        context.getString(R.string.singin_no_button),
+        (dialogInterface, i) ->
+            showDialogTask.setException(
+                new FirebaseAppDistributionException(
+                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
+
+    signInDialog.setOnCancelListener(
+        dialogInterface ->
+            showDialogTask.setException(
+                new FirebaseAppDistributionException(
+                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
+
+    signInDialog.show();
```

Comment:
I see we have a variable for updateDialog do we want something similar for signInDialog?

Reply:
I'm not sure I understand... could you point me to the line(s) of code?

Reply:
I think this question was answered below. Where we use updateDialogShown instead of the updateDialog.isShowing().